### PR TITLE
Esp8266

### PR DESCRIPTION
**Pull request by ejrennick**
log data skeleton

### CODE REVIEW

File: firmware/Core/Inc/Common.h [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 #define CTRL_LOOP_FREQUENCY 50  // Hz
 
+#define LOG_SUCCEEDED                 1
+#define LOG_FAILED                    0
```
Comment (Fberbari) [inline firmware/Core/Inc/Common.h:19]: Feel free to just use the global RETROFRIDGERATION_SUCCEEDED /FAILED macros. No need for each module to have it's own macros defined.

File: firmware/Core/Inc/Logs.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+*/
+void Logs_Init(void);
+
+/**
+* Logs data from sensors, passes to control module.
+* @param[in]        LogData          struct containing the average, min and max temperature
+* @return           LOG_SUCCEEDED or LOG_FAILED
+*/
+int Logs_LogData(const LogData_t *LogData);
+bool Logs_WifiData(const LogData_t *LogData);
```
Comment (Fberbari) [inline firmware/Core/Inc/Logs.h]: just keep these as int return types in case busy is a valid return type.

File: firmware/Core/Inc/Logs.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+bool Logs_FlashData(const LogData_t *LogData);
```
Comment (Fberbari) [inline firmware/Core/Inc/Logs.h]: Also need something to log to LCD right ? You should talk to Jacob about how to interact with his lower level LCD driver.

File: firmware/Core/Src/Logs.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * Code
+ **********************************************************************************************************************/
+
+void Logs_Init(void)
+{
+}
+
+int Logs_LogData(const LogData_t * LogData)
+{
+    if(Logs_WifiData(* LogData)):
```
Comment (Fberbari) [inline firmware/Core/Src/Logs.c]: these functions were exposed in the interface. But you also call them from the inside of the module. I'd recommend that you call Logs_WifiData, Logs_FlashData, and Logs_LCDData from your state in Controller.c Not a big deal, but will make things slightly more explicit and make the debug effort easier. Either that or don't expose those and only expose Logs_LogData.

As a rule, only expose functions meant to be called from the higher level in the Interface. Any helper functions part of your module should be declared static inside the .c file.

File: firmware/Core/Src/Logs.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return LOG_SUCCEEDED;
+    else:
+        return LOG_FAILED;
+}
+
+bool Logs_WifiData(const LogData_t * LogData)
+{
+  // check wifi connnection
+  // if wifi not connected, return 0
+  // Wifi_Init()
```
Comment (Fberbari) [inline firmware/Core/Src/Logs.c:21]: Shouldn't Wifi_init be called from the Logs_Init fuction ?

**Review by Fberbari — COMMENTED** [review]
Nothing retarded. Generally looks good.

File: firmware/Core/Src/Logs.c (hunk 5 above)
Comment (ejrennick) [reply firmware/Core/Src/Logs.c:21]: wifi_init starts the UART interrupt

File: firmware/Core/Inc/Logs.h (hunk 3 above)
Comment (ejrennick) [reply firmware/Core/Inc/Logs.h]: oops forgot about that

File: firmware/Core/Src/Logs.c (hunk 4 above)
Comment (ejrennick) [reply firmware/Core/Src/Logs.c]: oh ok makes more sense thx

File: firmware/Core/Src/Logs.c (hunk 5 above)
Comment (ejrennick) [reply firmware/Core/Src/Logs.c:21]: hmm

Comment (Fberbari) [reply firmware/Core/Src/Logs.c:21]: What should generally happen is the Init function, which gets called once, starts the UART interrupt. Then the processor just waits until the event occurs. After it does, and it's done processing the interrupt, the last thing you do before you exit the interrupt service routine is get it to start the interrupt again.

**Pull request closed without merging**